carp.common/src/commonMain/kotlin/dk/cachet/carp/common/application/sampling/AdaptiveGranularitySampling.kt

```diff
@@ -0,0 +1,33 @@
+@file:Suppress( "MatchingDeclarationName" )
+
+package dk.cachet.carp.common.application.sampling
+
+import dk.cachet.carp.common.application.data.DataType
+import dk.cachet.carp.common.application.devices.DeviceDescriptor
+
+
+/**
+ * Sampling scheme which provides only indirect control over how data is sampled by specifying a desired level of [Granularity].
+ * The levels of granularity correspond to expected degrees of power consumption.
+ * By default, [Granularity.Balanced] is used; when the battery is low, [Granularity.Coarse]; when battery is critically low, sampling stops.
+ */
+class AdaptiveGranularitySamplingScheme( dataType: DataType ) :
```

Comment:
I imagine an adaptive sampling scheme could adapt to something other than battery level? Just saying that this might be too generic a name for something that might have several specializations or be even more generalized in the future.

Reply:
Mmmm. I imagine other conditions would likely be modeled as `Trigger`s which start a `Measure` which overrides the `SamplingConfiguration`.

Regardless, I also thought this might be too generic when I wrote it, but found the longer alternative too convoluted. I.e., `BatteryAwareGranularitySamplingScheme`. Or maybe not? 🤷 

Suggestions?

Reply:
Maybe just `BatteryAwareSamplingScheme`. I think "aware" implies its adaptiveness. The granularity part sounds a bit convoluted :)